### PR TITLE
add rocm4.2 to nightly wheel builds

### PR DESCRIPTION
**Pull request by jeffdaily**

### CODE REVIEW

**Review by fmassa — COMMENTED** [review]
LGTM, but I'll let @malfet or @seemethere merge

**Review by seemethere — APPROVED** [review]

File: .circleci/regenerate.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def build_workflows(prefix='', filter_branch=None, upload=False, indentation=6,
     for btype in ["wheel", "conda"]:
         for os_type in ["linux", "macos", "win"]:
             python_versions = PYTHON_VERSIONS
-            cu_versions_dict = {"linux": ["cpu", "cu102", "cu111", "rocm4.0.1", "rocm4.1"],
+            cu_versions_dict = {"linux": ["cpu", "cu102", "cu111", "rocm4.0.1", "rocm4.1", "rocm4.2"],
```
Comment (malfet) [inline .circleci/regenerate.py:33]: Why do we need to keep 3 version of rocm here? Can we limit it down to just 2? (4.1 and 4.2 for example)

**Review by malfet — CHANGES_REQUESTED** [review]
Please remove older version of ROCm before adding new ones. (Or add a note why all 3 are still needed)

File: .circleci/regenerate.py (hunk 1 above)
Comment (jeffdaily) [reply .circleci/regenerate.py:33]: We at least need N and N-1.  I thought we were following the established tradition of keeping latest and two prior rocm versions.  Is this not the same for torchvision?  Let us know.  Like I said, keeping latest and one prior is useful.

Comment (jeffdaily) [reply .circleci/regenerate.py]: Done.  Removed rocm4.0.1 build in commit 5098b4331e2c7f929288517aea04629d513d8b82.

**Comment by jeffdaily** [comment]
@malfet I removed rocm 4.0.1.  Ready to merge?

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
@jeffdaily looks good to me, but I though that 4.1 is not going to be used in any releases, so 4.0.1 and 4.1 are they only two that should remain there?

**Pull request merged**